x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/TestCaseSupplier.java

```diff
@@ -117,6 +117,9 @@ public static List<TestCaseSupplier> stringCases(
     @Override
     public TestCase get() {
         TestCase supplied = supplier.get();
+        if (types.size() != supplied.getData().size()) {
```

Comment:
The `for` below wasn't checking the case where the TestCase data had more types than the supplier. This required fixing some cases in Top and Case